tests/Constraint/MultiConstraintTest.php

```diff
@@ -275,6 +275,20 @@ public function testCreatesMatchAllConstraintIfNoneGiven()
         $this->assertInstanceOf('Composer\Semver\Constraint\MatchAllConstraint', MultiConstraint::create(array()));
     }
 
+    public function testCreatesMatchAllConstraintIfConjunctiveAndCombinedWithAnotherONe()
+    {
+        $this->assertInstanceOf('Composer\Semver\Constraint\MatchAllConstraint', MultiConstraint::create(
+            array(new Constraint('>=', '2.5.0.0-dev'), new MatchAllConstraint())
+        ));
+    }
+
+    public function testCreatesMatchAllConstraintIfDisjunctiveAndCombinedWithAnotherONe()
+    {
+        $this->assertInstanceOf('Composer\Semver\Constraint\MatchAllConstraint', MultiConstraint::create(
+            array(new Constraint('>=', '2.5.0.0-dev'), new MatchAllConstraint()), false
```

Comment:
I think in this case it should return `MatchAllConstraint`, and above it should just drop the matchall as it is useless, i.e. `1.* && *` => `1.*`, `1.* || *` => `*`

Reply:
Updated.